sys/ztimer/periph_ptp.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C)    2018 Kaspar Schleiser <kaspar@schleiser.de>
+                    2021 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_ztimer_periph_ptp
+ * @{
+ *
+ * @file
+ * @brief       ztimer periph/ptp implementation
+ *
+ * @author      Jana Eisoldt <jana.eisoldt@ovgu.de>
+ *
+ * @}
+ */
+#include "assert.h"
+
+#include "irq.h"
+#include "periph/ptp.h"
+#include "ztimer/periph_ptp.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static ztimer_clock_t *clock_timer;
+
+void ptp_timer_cb(void)
+{
+    ztimer_handler(clock_timer);
+}
+
+static void _ztimer_periph_ptp_set(ztimer_clock_t *clock, uint32_t val)
+{
+    (void)clock;
+
+    unsigned state = irq_disable();
```

Comment:
```suggestion
```

Not needed, the PTP clock implementations are expected to disable IRQs, if needed. (And the only implementation we have does disable IRQs.)